tests/ssg_test_suite/xml_operations.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python2
+import xml.etree.cElementTree as ET
+
+import logging
+
+NAMESPACES = {
+    'xccdf': "http://checklists.nist.gov/xccdf/1.2",
+    'ds': "http://scap.nist.gov/schema/scap/source/1.2",
+    'xlink': "http://www.w3.org/1999/xlink",
+}
+
+
+logging.getLogger(__name__).addHandler(logging.NullHandler())
+
+
+def infer_benchmark_id_from_component_ref_id(datastream, ref_id):
+    root = ET.parse(datastream).getroot()
+    component_ref_node = root.find("*//ds:component-ref[@id='{0}']"
+                                   .format(ref_id), NAMESPACES)
+    if component_ref_node is None:
+        msg = (
+            'Component reference of Ref-Id {} not found within datastream'
+            .format(ref_id))
+        raise RuntimeError(msg)
+
+    comp_id = component_ref_node.get('{%s}href' % NAMESPACES['xlink'])
+    comp_id = comp_id.lstrip('#')
+    component_node = root.find("ds:component[@id='{0}']"
+                               .format(comp_id), NAMESPACES)
+    if component_node is None:
+        msg = 'Component of Id {} not found within datastream'.format(comp_id)
+        raise RuntimeError(msg)
+
+    benchmark_node = component_node.find("xccdf:Benchmark", NAMESPACES)
+    if benchmark_node is None:
+        msg = (
+            'Benchmark not found within comonent of Id {}'
+            .format(comp_id)
+        )
+        raise RuntimeError(msg)
+
+    return benchmark_node.get('id')
```

Comment:
I am not 100% sure whether ElementTree supports it but this can be done iwth just one XPath on maybe 3 lines of code.

```//cdf:Benchmark[parent::ds:component/@id=//ds:component-ref[@id=$ref_id]]/@id```

Untested and probably needs changes but I think it gets the idea across.

That said, I think this whole code can be avoided completely by changing the profile scraping code.

Reply:
I didn't know what is possible to achieve with the xpath language. This one-liner would be very impressive, but the xpath from python standard library doesn't support this as I get errors that indicate that `//` is not supported properly and then it throws `prefix 'parent' not found in prefix map` error (i.e. it doesn't understand XPath Axes).

Reply:
ACK. I looked into it and the python2 stdlib xpath is very limited.